### DISCUSSION TITLE
Lesson 8 html - window.etherium undefined error

### DISCUSSION THREAD

**Discussion by ediber** [Cyfrin/foundry-full-course-cu#147] [General]
i cloned the relevent project from git
i have metamask app runing
on inspet -> console when i type "window.etherium "
i get "undefined" 

i tried it on chrome and edge, on both cases netmask is running and have sepoila funds

**Comment by palmcivet7** [comment]
Try "window.ethereum"?

**Reply by ediber** [reply]
writing "etherium" was very supid mistake
but even when i write "window.ethereum" i get "undefined"
and i have a working turned on metamask estention
any ideas ?

**Reply by cromewar** [reply]
Please use a chromium based browser and try again @ediber

**Reply by ediber** [reply]
i googled the term "chromium"
it looks like chrome and edge are based on chromium
so i already tried it on  chromium based browser if im correct

**Reply by allwin199** [reply]
If you do window.ethereum on a new tab it will return "undefined".
Open a website like gmail.com, then do  window.ethereum
If it works in gmail.com, it will definitely work in your cloned app too.

**Reply by ediber** [reply]
in gmail window.ethereum results in
Proxy(c) {_events: {…}, _eventsCount: 0, _maxListeners: 100, _log: c, _state: {…}, …}

so there is a problem with my client code ?
because i only cloned it from the git of this course

`<!DOCTYPE html>
<html>
  <head>
    <title>Fund Me App</title>
  </head>

  <body>
    <button id="connectButton">Connect</button>
    <button id="balanceButton">getBalance</button>
    <button id="withdrawButton">Withdraw</button>
    
    <label for="ethAmount">ETH Amount</label>
    <input id="ethAmount" placeholder="0.1" />
    <button type="button" id="fundButton" > Fund </button>
    
  </body>

  <script src="./index.js" type="module"></script>
</html>
`

**Reply by allwin199** [reply]
After cloning, if you double click index.html file you will see all the buttons, in that page do window.ethereum

**Reply by ediber** [reply]
yes
thats the file im looking at
and here i get the error 

on inspet -> console when i type "window.etherium "
i get "undefined"

**Reply by PatrickAlphaC** [reply]
type `window.ethereum` not `window.etherium`

**Reply by SuperHumano** [reply]
for me that's the same thing, i type window.ethereum and i get "undefined", i click in the button connect and nothing happens, i changed the network than, from sepolia to mainnet and still the same thing . There is two errors that also appear in the console for me, maybe, probably have something to do with:
![error_to_solve](https://github.com/Cyfrin/foundry-full-course-f23/assets/127708514/803b3253-7e7e-41da-97f8-346ed5da9f92)

**Reply by SuperHumano** [reply]
![error_solved](https://github.com/Cyfrin/foundry-full-course-f23/assets/127708514/50a57011-4c79-4cc5-9b1b-e90cf4328f67)

Now, for me is working, I was travelling through discussions and saw that the live server for me wasn't functioning as expected, but with the extention Live Server (Five Server) it worked out

**Reply by Shoolapani** [reply]
@SuperHumano  Thanks, it also worked for me, was facing the same issue.

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/83538929/82ba5d3d-25db-4664-82a7-a315e23a70a7)
